src/main/java/carparks/parking/FeeCalculator.java

```diff
@@ -0,0 +1,56 @@
+package carparks.parking;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+@Component
+class FeeCalculator {
+
+	public BigDecimal calculateFee(ParkingEntry entry) {
+
+		LocalDateTime start = entry.getStart();
+		LocalDateTime end = (entry.getFinish() == null) ? LocalDateTime.now() : entry.getFinish();
```

Comment:
W jaki sposób przetestowałbyś sytuację z `entry.getFinish() == null`? Jakbyś zamockował `now()`?

Reply:
Faktycznie, wydawało mi się, że ważna jest dla nas tylko różnica między start a finish - i dlatego w tych testach, które napisałem nie trzeba było mockować now(), bo ten niedeterminizm nam nie przeszkadzał, ale teraz faktycznie widzę, że w przypadku gdy entry.getFinish == null, to robi się problem. Żeby go rozwiązać można albo albo użyć obiektu Clock, i w teście ręcznie nadpisywać wartość zwracaną przez now(), albo ukryć pobieranie czasu za interfesjem i na produkcji zwracać normalnie now(), a w testach mockować i uczyć tę metodę co ma zwracać, czyli konkretny czas.

Reply:
Zgadza się, ogólnie dobrą praktyką jest mieć wywołanie `now()` tylko raz w projekcie